components/cli/pkg/commands/instance/route_traffic.go

```diff
@@ -22,79 +22,84 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/cellery-io/sdk/components/cli/cli"
 	errorpkg "github.com/cellery-io/sdk/components/cli/pkg/error"
-	"github.com/cellery-io/sdk/components/cli/pkg/kubernetes"
 	"github.com/cellery-io/sdk/components/cli/pkg/routing"
 	"github.com/cellery-io/sdk/components/cli/pkg/util"
 )
 
-func RunRouteTrafficCommand(sourceInstances []string, dependencyInstance string, targetInstance string, percentage int,
+func RunRouteTrafficCommand(cli cli.Cli, sourceInstances []string, dependencyInstance string, targetInstance string, percentage int,
 	enableUserBasedSessionAwareness bool, assumeYes bool) error {
-	spinner := util.StartNewSpinner(fmt.Sprintf("Starting to route %d%% of traffic to instance %s", percentage,
+	var err error
+	artifactFile := fmt.Sprintf("./%s-routing-artifacts.yaml", dependencyInstance)
+	defer func() error {
+		return os.Remove(artifactFile)
+	}()
+	BuildRouteArtifact(cli, sourceInstances, dependencyInstance, targetInstance, percentage,
+		enableUserBasedSessionAwareness, assumeYes)
+
+	if err = cli.ExecuteTask("Applying modified rules", "Failed to apply modified rules", "", func() error {
+		err = cli.KubeCli().ApplyFile(artifactFile)
+		if err != nil {
+			return fmt.Errorf("error occurred while applying modified rules, %v", err)
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	util.PrintSuccessMessage(fmt.Sprintf("Successfully routed %d%% of traffic to instance %s", percentage,
+		targetInstance))
+	return nil
+}
+
+func BuildRouteArtifact(cli cli.Cli, sourceInstances []string, dependencyInstance string, targetInstance string, percentage int,
```

Comment:
Do we need to make this function accessible from outside?

Reply:
Updated.